src/components/TelegramConfiguringOverlay.tsx

```
@@ -53,11 +56,15 @@ export default function TelegramConfiguringOverlay({
   const [phase, setPhase] = useState(0);
   const [dots, setDots] = useState("");
   const overlayRef = useRef<HTMLDivElement>(null);
-  const cancelledRef = useRef(false);
+  const onDoneRef = useRef(onDone);
+  const onTimeoutRef = useRef(onTimeout);
+  onDoneRef.current = onDone;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/components/TelegramConfiguringOverlay.tsx:61:3
  59 |   const onDoneRef = useRef(onDone);
  60 |   const onTimeoutRef = useRef(onTimeout);
> 61 |   onDoneRef.current = onDone;
     |   ^^^^^^^^^^^^^^^^^ Cannot update ref during render
  62 |   onTimeoutRef.current = onTimeout;
  63 |
  64 |   useEffect(() => {

(react-hooks/refs)